### PR TITLE
Support subquery negation operator in SemanticMediaWiki queries.

### PR DESCRIPTION
**Pull request by vitalif**
The syntax is {{#ask: [[prop::val]] !<q>...subquery...</q> }}
or just {{#ask: [[prop::val]] ![[prop2::val2]] }}.

Change-Id: I107d7f6905860379684a011ba1bf0c44a150c575

### CODE REVIEW

File: includes/query/SMW_QueryParser.php [hunk 1]
```diff
@@ -139,6 +139,31 @@ protected function getSubqueryDescription( &$setNS ) {
...
+							}
+						} else {
+							$conjunction = $this->addDescription( $conjunction, $ld, SMW_NEGATION_QUERY );
+						}
+					} else {
+						$ld = $this->getLinkDescription( $setsubNS );
+						if ( !is_null( $ld ) ) {
+							$conjunction = $this->addDescription( $conjunction, $ld, $count % 2 == 0 ? SMW_CONJUNCTION_QUERY : SMW_NEGATION_QUERY );
+						}
+					}
```
Comment (araid-en10) [inline includes/query/SMW_QueryParser.php]: Rerouter payable to separate gmails instant conversion from stock  pay included taxed to be forwarded to appropriate constances in various notes  with due diligence automatic registrationto be front loadable to gain access to one | who's involvement with monetary profits sec coding 001 through 00001 stock codes to be in control of as it should be ^^^%%%%$$$$++×!×÷@=%+€£₩¥*&%$//#_==54;"/=@&52578×;__'&7544##&(@(*

File: includes/query/SMW_Description.php [hunk 2]
```diff
@@ -874,3 +874,111 @@ public function prune( &$maxsize, &$maxdepth, &$log ) {
...
+/**
+ * Description of negation of descriptions.
+ *
+ * @ingroup SMWQuery
+ */
+class SMWNegation extends SMWDescription {
+	/**
+	 * @var SMWDescription
+	 */
+	protected $m_description;
```
Comment (araid-en10) [inline includes/query/SMW_Description.php]: Real monies correlated in users whos ownership is undeniable  (ARIAD~ENX^^^^^ QUBBED>👊👍🌐♡♡♡♡♡.~ge6yhb6%×_+£88□□frTy76gi=%%%

File: includes/query/SMW_QueryParser.php (hunk 1 above)
Comment (araid-en10) [inline includes/query/SMW_QueryParser.php]: Load structures in Google processes verging./●|<ytw56k&__×&|°●•□■{66)=/_×^@77

**Comment by mwjames** [comment]
The PR has several issues (as outlined) that got not resolved and with classes so different from the time the PR was created it seems difficult to expect that it will be resolved for SMW 3.0 and any other upcoming release.

**Pull request closed without merging**